### ISSUE TITLE
server lisenting on 0.0.0.0:9020/9022

### ISSUE THREAD

**Issue by kanexu1985** [kungfoo/falcon-bms-control#28]
Hi kungfoo, first great thanks for the amazing idea and app, but when I start the windows server, it shows:
```
2023-02-23 07:57:35.2920|INFO|FalconBmsUniversalServer|Starting up...
2023-02-23 07:57:35.4080|INFO|FalconBmsUniversalServer|Listening for broadcast packets on: 0.0.0.0:9020
2023-02-23 07:57:35.4270|INFO|FalconBmsUniversalServer|Running on 0.0.0.0:9022
```

and my client device cannot get connected, only keep showing
```
Connecting to 192.168.x.x.....
```
(I've manually set the static IP which my PC is)

shouldn't it be my PC IP like 192.168.x.x, rather than 0.0.0.0? I open the .cfg file and can't find any section of IP settings.
Could you pls help?

I'm on Win11, not sure if that matters?

Thanks

**Comment by kungfoo** [comment]
`0.0.0.0` is a special address used to listen on all interfaces, if a
computer has more than one. This is correct. Make sure you allow the server
app to receive traffic on port 9022 in windows firewall.

Most times windows firewall prompts for allowing it the first time you run
a program. After that you have to go into the actual settings dialog.

**Comment by kanexu1985** [comment]
thanks kungfoo for your quick reply.

Yes I of course allow the Windows firewall on the promts dialog when the first time I run the program. I also double checked it on the firewall settings and it is allowing the program.

I also clicked the "allow discovering other devices in network" on my ipad the first time launching the app(ipadOS 14.8.1)

However, on my ipad, it just doesn't get connected, either I leave the "Server IP" field blank or specified, it's not getting connected.

Am I doing the steps in right sequence? here's what I tried:
1. start falcon-bms-control-server.exe
2. launch ipad app
3. start BMS game, and take a flight

with or without the 3rd step, nothing is changing, ipad app is not connecting. I'm sure the ipad and the PC are under the same network, other app like VNC viewer can access the PC.

**Comment by kungfoo** [comment]
There is no specific order the steps need to be taken in,  all orders
should work. However, i also do them in this order, usually.

Can you upgrade iOS on that device? I'm not sure i remember correctly,  but
there was one version of iOS that had problems with udp.

**Comment by kanexu1985** [comment]
hi Kungfoo,

I tried again today and now it worked.

The iOS device I bought is from China, and there's a special option intergraded in the iOS on the device sold from China. That is, when an APP first time attempt to get network traffic, there'll be a pop-up menu letting user to select:
- allow network traffic over cellur & wifi
- allow network traffic over wifi only 
- not allow network traffic

if this pop-up wasn't shown, iOS will block the network by default of the APP. And somehow, from your APP, this menu wasn't shown, thus iOS is blocking its network. But today when I manually typed the IP, the menu shows up, and after I allow it, it just worked.

Thanks anyway for your replies, hope my case may provide some experience in future if anyone has same problem like mine.

:-)
